vortex-file/src/read/mask.rs

```diff
@@ -105,29 +128,35 @@ impl RowMask {
                 return Err(err());
             }
 
-            let mut bitmap = Bitmap::new();
+            let mut buffer_builder = BooleanBufferBuilder::new_from_buffer(MutableBuffer::new((end - begin).div_ceil(8)), end - begin);
 
             iterate_integer_array!(array, |$P, $iterator| {
                 for batch in $iterator {
                     for index in batch.data() {
-                        bitmap.add(u32::try_from(*index).map_err(|_| err())?);
+                        bit_util::set_bit(buffer_builder.as_slice_mut(), *index as usize)
```

Comment:
`buffer_builder.set_bit(*index as usize, true)`

Reply:
the whole idea was that this is always true so I can call underlying method

Reply:
the method is inline anyway

Reply:
fair, I didn't see the annotation